pkg/controller/storage/capability/capability_controller.go

```diff
@@ -208,13 +208,17 @@ func (c *StorageCapabilityController) syncHandler(key string) error {
 
 	// Annotate storageClass
 	storageClassUpdated := storageClass.DeepCopy()
-	err = c.addStorageClassSnapshotAnnotation(storageClassUpdated, isCSIStorage)
-	if err != nil {
-		return err
-	}
-	err = c.addCloneVolumeAnnotation(storageClassUpdated, isCSIStorage)
-	if err != nil {
-		return err
+	if !isCSIStorage {
+		c.removeAnnotations(storageClassUpdated)
+	} else {
+		err = c.updateStorageClassSnapshotAnnotation(storageClassUpdated, isCSIStorage)
+		if err != nil {
+			return err
+		}
+		err = c.updateCloneVolumeAnnotation(storageClassUpdated, isCSIStorage)
+		if err != nil {
+			return err
+		}
 	}
 	_, err = c.storageClassClient.Update(context.Background(), storageClassUpdated, metav1.UpdateOptions{})
```

Comment:
it seems you did not compare the annotations of `storageClassUpdated` and `storageClass`, your code will update the storageclass every time. Is that okay?

Reply:
I parsed the annotations in the update function, if it can be parsed as bool, return directly.Otherwise it will be updated.